presto-main/src/main/java/com/facebook/presto/server/security/AuthenticationFilter.java

```diff
@@ -126,6 +141,43 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         }
     }
 
+    public CustomHttpServletRequestWrapper mergeExtraHeaders(HttpServletRequest request, Principal principal)
+    {
+        CustomHttpServletRequestWrapper wrappedRequest = new CustomHttpServletRequestWrapper(request);
+        Map<String, String> extraHeadersMap = new HashMap<>();
+        Set<String> globallyAddedHeaders = new HashSet<>();
```

Comment:
Please use immutable map builders and construct when needed.

Reply:
I have used immutable map builders.